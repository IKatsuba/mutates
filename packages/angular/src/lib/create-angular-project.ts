import { Tree } from '@angular-devkit/schematics';
import { ProjectOptions } from 'ts-morph';

import { createProject } from '@mutates/core';

import { NgTreeFileSystem } from './ng-tree-file-system';

export function createAngularProject(tree: Tree, options?: Omit<ProjectOptions, 'fileSystem'>) {
  return createProject(new NgTreeFileSystem(tree), options);
}
