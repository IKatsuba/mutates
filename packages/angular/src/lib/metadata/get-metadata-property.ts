import { ClassDeclaration, PropertyAssignment } from 'ts-morph';

import { getObjectProperties } from '@mutates/core';

import { MetadataProperty } from '../types/metadata-property';
import { getMetadata } from './get-metadata';

export function getMetadataProperty<T extends MetadataProperty>(
  klass: ClassDeclaration,
  property: T,
): PropertyAssignment | undefined {
  const [metadata] = getMetadata(klass);

  if (!metadata) {
    return;
  }

  return getObjectProperties(metadata, {
    name: property as string,
  }).at(0);
}
