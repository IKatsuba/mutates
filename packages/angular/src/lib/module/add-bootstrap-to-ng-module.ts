import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addBootstrapToNgModule(
  classDeclaration: ClassDeclaration,
  component: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'NgModule', 'bootstrap', component, {
    unique,
    forceToArray: true,
  });
}
