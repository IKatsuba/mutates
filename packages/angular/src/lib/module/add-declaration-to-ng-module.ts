import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addDeclarationToNgModule(
  classDeclaration: ClassDeclaration,
  declaration: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'NgModule', 'declarations', declaration, {
    unique,
    forceToArray: true,
  });
}
