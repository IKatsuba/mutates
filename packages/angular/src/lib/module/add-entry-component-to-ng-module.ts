import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addEntryComponentToNgModule(
  classDeclaration: ClassDeclaration,
  component: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'NgModule', 'entryComponents', component, {
    unique,
    forceToArray: true,
  });
}
