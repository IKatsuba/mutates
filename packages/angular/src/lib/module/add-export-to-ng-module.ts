import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addExportToNgModule(
  classDeclaration: ClassDeclaration,
  exportName: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'NgModule', 'exports', exportName, {
    unique,
    forceToArray: true,
  });
}
