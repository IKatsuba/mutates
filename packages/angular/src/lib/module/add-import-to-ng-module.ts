import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addImportToNgModule(
  classDeclaration: ClassDeclaration,
  moduleName: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'NgModule', 'imports', moduleName, {
    unique,
    forceToArray: true,
  });
}
