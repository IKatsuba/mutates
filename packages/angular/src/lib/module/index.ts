export * from './add-bootstrap-to-ng-module';
export * from './add-declaration-to-ng-module';
export * from './add-entry-component-to-ng-module';
export * from './add-export-to-ng-module';
export * from './add-import-to-ng-module';
export * from './add-provider-to-ng-module';
export * from './add-schema-to-ng-module';
