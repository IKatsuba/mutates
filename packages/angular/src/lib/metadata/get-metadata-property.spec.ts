import { Tree } from '@angular-devkit/schematics';

import { createSourceFile } from '@mutates/core';

import { createAngularProject } from '../create-angular-project';
import { getNgModule } from '../module';
import { createTestingTree } from '../testing';
import { getMetadataProperty } from './get-metadata-property';

describe('getMetadataProperty', () => {
  let host: Tree;
  beforeEach(() => {
    host = createTestingTree();

    createAngularProject(host);
  });

  it('should get metadata property', () => {
    createSourceFile(
      'src/main.ts',
      `
@NgModule({providers: []}) class AppModule {}
    `,
    );
    const metadataProperty = getMetadataProperty(getNgModule('src/main.ts').at(0)!, 'providers');

    expect(metadataProperty?.getText()).matchSnapshot();
  });
});
