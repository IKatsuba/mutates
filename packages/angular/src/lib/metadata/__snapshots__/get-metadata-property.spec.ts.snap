// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getMetadataProperty > should get metadata property 1`] = `"providers: []"`;
